packages/functional-tests/tests/key-stretching-v2/signInTokenCode.spec.ts

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Page, expect, test } from '../../lib/fixtures/standard';
+import { BaseTarget } from '../../lib/targets/base';
+import { SettingsPage } from '../../pages/settings';
+import { DeleteAccountPage } from '../../pages/settings/deleteAccount';
+
+/**
+ * These tests represent various permutations between interacting with V1 and V2
+ * key stretched passwords. We need to ensure that operations are interchangeable!
+ */
+test.describe('severity-2 #smoke', () => {
+  type Version = { version: 1 | 2; query: string };
+  type TestCase = { signinVersion: Version };
+  const v1: Version = { version: 1, query: '' };
+  const v2: Version = { version: 2, query: 'stretch=2' };
+  const TestCases: TestCase[] = [{ signinVersion: v1 }, { signinVersion: v2 }];
+
+  for (const { signinVersion } of TestCases) {
+    test(`fffff signs up as v1, sign in within token code, signs  as v${signinVersion.version}`, async ({
+      page,
+      target,
+      pages: {
+        settings,
+        signin,
+        connectAnotherDevice,
+        deleteAccount,
+        signinTokenCode,
+      },
+      testAccountTracker,
+    }, {}) => {
+      const { email, password } = await testAccountTracker.signUpSync();
+      await page.goto(
+        `${target.contentServerUrl}?context=fx_desktop_v3&service=sync&action=email&${signinVersion.query}`
+      );
+      await signin.fillOutEmailFirstForm(email);
+      await signin.fillOutPasswordForm(password);
+
+      await expect(page).toHaveURL(/signin_token_code/);
+      await expect(signinTokenCode.heading).toBeVisible();
+      const code = await target.emailClient.getVerifyLoginCode(email);
+      await signinTokenCode.fillOutCodeForm(code);
+
+      await expect(page).toHaveURL(/pair/);
+      await connectAnotherDevice.clickNotNowPair();
+
+      await expect(page).toHaveURL(/settings/);
+
+      // Make sure upgrade occurred
+      if (signinVersion.version === 2) {
+        const client = target.createAuthClient(2);
+        const status = await client.getCredentialStatusV2(email);
+        expect(status.currentVersion).toEqual(`v2`);
+      }
+
+      // Delete blocked account, required before teardown
```

Comment:
Is the account blocked?

Reply:
Nope. Just some copy and pasta. I'll fix that.